src/Adapter/Product/Stock/Update/ProductStockUpdater.php

```diff
@@ -305,23 +322,34 @@ private function updateStockAvailable(StockAvailable $stockAvailable, ProductSto
         $this->stockAvailableRepository->update($stockAvailable);
 
         if ($properties->getStockModification()) {
+            if ($stockAvailable->getShopId()) {
+                $shopId = $stockAvailable->getShopId();
+            } else {
+                // We can use any shop from the shop group, the stock movement will correctly associate the proper StockAvailable based on the shopId only
+                $shopsFromGroup = $this->shopGroupRepository->getShopsFromGroup(new ShopGroupId((int) $stockAvailable->id_shop_group));
+                $shopId = $shopsFromGroup[0]->getValue();
```

Comment:
If you want to fetch the first item I think you should use reset() instead since we don't have the control on the index type

Reply:
Hmm, but we dot know how the DB result is returned though 🤔 It's always a 0-indexed array